auth.js

```diff
@@ -223,158 +216,155 @@ module.exports = function auth( options ) {
-        return do_respond( err, 'register', req, respond )
+      if (err) {
+        return do_respond(err, 'register', req, respond)
       }
 
-      seneca.act( "role:'user',cmd:'register'", details , function( err, out ) {
-        if( err || !out.ok ) {
-          return do_respond( err || (out ? out.why: 'Internal server error'), 'register', req, respond )
+      seneca.act("role:'user',cmd:'register'", details, function (err, out) {
+        if (err || !out.ok) {
+          return do_respond(err || (out ? out.why : 'Internal server error'), 'register', req, respond)
         }
 
-        seneca.act( "role:'user',cmd:'login'", { nick:out.user.nick, auto:true }, function( err, out ) {
-          if( err || !out.ok ) {
-            return do_respond( err || (out ? out.why: 'Internal server error'), 'register', req, respond )
+        seneca.act("role:'user',cmd:'login'", {nick: out.user.nick, auto: true}, function (err, out) {
+          if (err || !out.ok) {
+            return do_respond(err || (out ? out.why : 'Internal server error'), 'register', req, respond)
           }
 
-          if( req && req.seneca ) {
-            req.seneca.user  = out.user
+          if (req && req.seneca) {
+            req.seneca.user = out.user
             req.seneca.login = out.login
 
-            req.seneca.act( "role: 'auth', set: 'token'", {
-              tokenkey: options.tokenkey, token: ( req.seneca.login.token || req.seneca.login.id )
-            }, function( err ) {
-              return do_respond( err, 'register', req, respond )
+            req.seneca.act("role: 'auth', set: 'token'", {
+              tokenkey: options.tokenkey, token: (req.seneca.login.token || req.seneca.login.id)
+            }, function (err) {
+              return do_respond(err, 'register', req, respond)
             })
           }
-          else{
+          else {
             respond(null, {
-              ok:    out.ok,
-              user:  out.user,
+              ok: out.ok,
+              user: out.user,
               login: out.login
-            } )
+            })
           }
         })
       })
     })
   }
 
+  function cmd_create_reset (msg, respond) {
+    seneca.act("role: 'auth', hook: 'map_fields'", {action: 'create_reset', data: msg.data}, function (err, userData) {
+      if (err) {
+        return respond(err, userData)
+      }
 
-  function cmd_create_reset( msg, respond ) {
-    seneca.act( "role: 'auth', hook: 'map_fields'", {action: 'create_reset', data: msg.data}, function( err, userData ) {
-      var nick  = userData.nick
+      var nick = userData.nick
       var email = userData.email
 
       var args = {}
-      if( void 0 != nick )  args.nick  = nick;
-      if( void 0 != email ) args.email = email;
+      if (void 0 !== nick) args.nick = nick
+      if (void 0 !== email) args.email = email
 
-      seneca.act( "role:'user',cmd:'create_reset'", args , respond )
+      seneca.act("role:'user',cmd:'create_reset'", args, respond)
```

Comment:
@mihaidma 

Here I think we need to change and sending the token and as part of response.